### ISSUE TITLE
GiftedChat cDU comparison does not allow you to update messages

### ISSUE THREAD

**Issue by audiolion** [FaridSafi/react-native-gifted-chat#1434]
#### Issue Description

In https://github.com/FaridSafi/react-native-gifted-chat/commit/ac984958c3761c15a9a1ebc6c6fca8d49ecd2bec the `componentDidUpdate` comparison for messages was changed, however the comparison no longer allows us to update messages.

#### Steps to Reproduce / Code Snippets

e.g.

-> user sends message

```js
// messages array passed to <GiftedChat />
[{ id: 1, status: 'SENDING' }]
```

-> websocket subscription updates message status

```js
// messages array passed to <GiftedChat />
[{ id: 1, status: 'DELIVERED' }]
```

The `componentDidUpdate` only is checking if the array lengths are still the same, which they are, and so it does not update and my `renderBubble` method is not called.

#### Expected Results

The `componentDidUpdate` should either a) check if contents of messages array has changed, or b) allow the user an opt-in like the [shouldMessageUpdate](https://github.com/FaridSafi/react-native-gifted-chat/blob/3cbadaf07ba683d160b35d6a9c42c2204ce48886/src/Message.tsx#L99) function that is passed to `Message` component

#### Additional Information

* Nodejs version: 10.16.0
* React version: 16.9.0
* React Native version: 0.61.x
* react-native-gifted-chat version: 0.10.0
* Platform(s) (iOS, Android, or both?): both
* TypeScript version: 3.6.3

**Comment by ucallQA** [comment]
I was just about to post this issue! We are having the same issue (for me in was checking the update in `renderTime`) - as soon as we updated from GiftedChat version `0.9.11` we were experiencing this. After many days of trouble shooting and making sure that our redux state was being deep cloned properly I figured it was ONLY checking for the length of the messages array since updating the messages was not re-rendering until the length of messages changed.

My notes from the issue: 
- Deleting a message will cause re-render
- Adding a new message will cause re-render
- Deleting a message and adding a new one at the same time will not cause a re-render (messages array length stays the same)

I was running out of options during our debugging process - if this could be changed back to work the way it did in `0.9.11` or some toggle to let us manually cause a re-render that would be awesome

Thank you
